### DISCUSSION TITLE
Using a reactive javascript sql result in a mosaic plot within Observable Framework

### DISCUSSION THREAD

**Discussion by oscaro00** [uwdata/mosaic#559] [Q&A]
Within the starter Framework app with the launches dataset, I am trying to create a reactive plot based on user's filter selection. The following logic works to filter and display the table.

```js
const test_query = sql([`
SELECT * FROM launches
  WHERE state IN ('--'${", ?".repeat(selectCountryTest.length)})
`], ...selectCountryTest);
```

```js
Inputs.table(test_query)
```

However, the following code throws an error:

```js
vg.plot(
  vg.barX(
    vg.from(test_query),
    {x: vg.count(), y: "family", fill: "#ccc", fillOpacity: 0.2}
  ),
  vg.barX(
    vg.from(test_query, {filterBy: $click}),
    {x: vg.count(), y: "family", fill: "#ccc"}
  ),
  vg.toggleY({as: $click}),
  vg.highlight({by: $click}),
  vg.xDomain(vg.Fixed),
  vg.yDomain($domain),
  vg.yLabel(null),
  vg.colorDomain($domain),
  vg.colorRange($colors),
  vg.marginLeft(45),
  vg.width(660)
)
```

If I replace, test_query with "launches" in the above code block, then the plot renders, but doesn't react to a user's selection.

The browser debugger reports the following error: "Uncaught (in promise) RuntimeError: index out of bounds"

Is test_query not the right type for the Mosaic plot function?

**Comment by domoritz** [comment]
`from` expected the table name. You can register the dataset with the coordinator.

**Reply by oscaro00** [reply]
I appreciate the help. From reading the DuckDB documentation on the Observable Framework website, there is an example of registering a table with the code

```js
const db = DuckDBClient.of({gaia: FileAttachment("gaia-sample.parquet")});
```

The documentation says `DuckDBClient.of()` works with arrow tables, but I'm having trouble with the syntax. Both 

```js
const db = DuckDBClient.of({interactive_tbl: Arrow.tableFromIPC(test_query)});
```
and
```js
const db = DuckDBClient.of({interactive_tbl: test_query});
```

are throwing errors. Am I missing something obvious?

**Reply by Fil** [reply]
duplicate of https://github.com/observablehq/framework/discussions/1747

**Reply by Fil** [reply]
I gave an answer here https://github.com/observablehq/framework/discussions/1747#discussioncomment-10949926
